### ISSUE TITLE
[Bug]:  The peer connetion can't be established with the error "IceAgent failed with 0x5a00000d" sometimes.

### ISSUE THREAD

**Issue by Teresa7755** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1890]
### Please confirm you have already done the following

- [X] I have searched the repository for related/existing bug reports
- [X] I have all the details the issue requires

### Describe the bug

The peer connetion can't be established with the error "IceAgent failed with 0x5a00000d" sometimes, even the network between my phone and embedded device are both fine and the offer/answer SDP and ICE candidates are all correct.

### Expected Behavior

The peer connetion should be established and users can play the live streaming if the network are both fine.

### Current Behavior

If a Viewer starts the live view, some of ICE candidates usually send to the Master faster than the offer SDP, such as the host or srflx type of candidates, and the live view works fine.  
The issue happens when ICE candidates are sent to the Master later than offer SDP, and the live view faild at this time. Please check the master log I attached.
(Please ignore errors "writeVideoFrame/writeAudioFrame failed with 0x5b000006" in sendMediaPackets in the log, they were caused by the peer connection was not established but we still sends the video and audio frames.)

[live_streaming_failed(Master).log](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/13793876/live_streaming_failed.Master.log)

### Reproduction Steps

Please try to make viewer sends the offer SDP first, then sends ICE candidates to master, the probability of the error "IceAgent failed with 0x5a00000d" occurring is 70% under that corner case. 

### WebRTC C SDK version being used

1.7.3

### Compiler and Version used

-

### Operating System and version

Linux

### Platform being used

-

**Comment by disa6302** [comment]
@Teresa7755 ,

Looking at the logs, it seems TURN is unable to create permissions for the relay candidates within the 2 second timeout. I also see that the remote relay candidates come in very late (after TURN failure). You mentioned when the candidates come in before SDP offer, peer connection is established. Can you confirm what candidates are being chosen? In the non-working case, what is the change in the set up? Where is the viewer running and what SDK is the viewer using?

One more note: you are using version 1.7.3 which is more than 1.5 years old. Could you switch to the most recent version of the SDK and run the application again? If encountering the same issue after the update, please provide VERBOSE complete logs. The logs you attached only includes logs SDP_OFFER onwards.

**Comment by niyatim23** [comment]
I'm seeing `0x5b000006` which is `STATUS_SRTP_NOT_READY_YET` which is pretty rare. Also, I see `sendMediaPackets` throwing this and it is not a part of our sample application. Can you run the unmodified sample application from the latest version of our SDK and see if you run into this issue?

**Comment by Teresa7755** [comment]
@niyatim23 Please ignore errors "writeVideoFrame/writeAudioFrame failed with 0x5b000006" in sendMediaPackets in the log, they were caused by the peer connection was not established but we still sends the video and audio frames.  It doesn't have impact on the peer connection of webrtc. But yes, we can try the unmodified sample application.

**Comment by niyatim23** [comment]
Sure, we are waiting on VERBOSE logs from the unmodified sample of our latest release
